docs/csharp/programming-guide/generics/constraints-on-type-parameters.md

```diff
@@ -10,72 +10,111 @@ helpviewer_keywords:
-## See Also  
- <xref:System.Collections.Generic>  
+
+When you define a generic class, you can apply restrictions to the kinds of types that client code can use for type arguments when it instantiates your class. If client code tries to instantiate your class by using a type that is not allowed by a constraint, the result is a compile-time error. These restrictions are called constraints. Constraints are specified by using the `where` contextual keyword. The following table lists the six types of constraints:
+
+|Constraint|Description|
+|----------------|-----------------|
+|`where T: struct`|The type argument must be a value type. Any value type except <xref:System.Nullable> can be specified. See [Using Nullable Types](../nullable-types/using-nullable-types.md) for more information.|
+|`where T : class`|The type argument must be a reference type; this applies also to any class, interface, delegate, or array type.|
+|`where T : unmanaged`|The type argument must not be a reference type and must not contain any reference type members at any level of nesting.|
+|`where T : new()`|The type argument must have a public parameterless constructor. When used together with other constraints, the `new()` constraint must be specified last.|
+|`where T :` *\<base class name>*|The type argument must be or derive from the specified base class.|
+|`where T :` *\<interface name>*|The type argument must be or implement the specified interface. Multiple interface constraints can be specified. The constraining interface can also be generic.|
+|`where T : U`|The type argument supplied for T must be or derive from the argument supplied for U.|
+
+Some of the constraints are mutually exclusive. All value types must have an accessible parameterless constructor. Therefore, the `struct` constraint implies the `new()` constraint and the `new()` constraint cannot be combined with the `struct` constraint. The `unmanaged` contraint implies the `struct` constraint. Therefore, the `unmanaged` constraint cannot be combined with either the `struct` or `new()` constraints.
+
+## Why Use Constraints
+
+If you want to examine an item in a generic list to determine whether it is valid or to compare it to some other item, the compiler must have some guarantee that the operator or method it has to call will be supported by any type argument that might be specified by client code. This guarantee is obtained by applying one or more constraints to your generic class definition. For example, the base class constraint tells the compiler that only objects of this type or derived from this type will be used as type arguments. Once the compiler has this guarantee, it can allow methods of that type to be called in the generic class. Constraints are applied by using the contextual keyword `where`. The following code example demonstrates the functionality we can add to the `GenericList<T>` class (in [Introduction to Generics](introduction-to-generics.md)) by applying a base class constraint.
```

Comment:
The sentence at the end of the paragraph
>Constraints are applied by using the contextual keyword `where`

is repeated from the first paragraph.

The first sentence is not very clear. (Though, it's not in the scope of this PR.)